### PR TITLE
[ISSUE #4157]Prevent the first time from failing to load

### PR DESCRIPTION
**Pull request by wzhangNJ**
Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by wzhangNJ** [comment]
Prevent the first time from failing to load，Sometimes it cannot be loaded for the first time，Such as SpringClouldFeign.
https://github.com/apache/shenyu/issues/4157

File: shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/client/AbstractContextRefreshedEventListener.java [hunk 1]
```diff
@@ -98,14 +98,14 @@ public AbstractContextRefreshedEventListener(final PropertiesConfig clientConfig
...
     public void onApplicationEvent(@NonNull final ContextRefreshedEvent event) {
-        if (!registered.compareAndSet(false, true)) {
-            return;
-        }
         final ApplicationContext context = event.getApplicationContext();
         Map<String, T> beans = getBeans(context);
         if (MapUtils.isEmpty(beans)) {
             return;
         }
+        if (!registered.compareAndSet(false, true)) {
```
Comment (yu199195) [inline shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/client/AbstractContextRefreshedEventListener.java:106]: why modify this？ 

when onApplicationEvent， getBeans(context) is null？

**Review by yu199195 — APPROVED** [review]

File: shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/client/AbstractContextRefreshedEventListener.java (hunk 1 above)
Comment (wzhangNJ) [reply shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/client/AbstractContextRefreshedEventListener.java:106]: yes

**Comment by dragon-zhang** [comment]
@wzhangNJ please add wechat Z18179469818

**Comment by wzhangNJ** [comment]
@dragon-zhang ok

**Pull request merged**